src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp

```diff
@@ -111,7 +111,11 @@ void TCPEndpoint::fromUriRaw(const bsl::string& uri)
 
     const size_t separator = uri.find_last_of(':');
 
-    d_port = bsl::strtol(uri.c_str() + separator + 1, 0, 10);
+    const long port = bsl::strtol(uri.c_str() + separator + 1, 0, 10);
+    if (port >= 0 && port <= 65535) {
+        // Maintaining strtol long value and casting if port is in range
+        d_port = static_cast<int>(port);
+    }
```

Comment:
Here, let's do the same and add a check:

```suggestion
    if (separator == bsl::string::npos) {
        return;  // RETURN
    }

    const long port = bsl::strtol(uri.c_str() + separator + 1, 0, 10);
    // For simplicity, do not accept ambiguous `port` value 0
    if (port <= 0 || port > 65535) {
        return;  // RETURN
    }
    
    d_port = static_cast<int>(port);
```